cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -633,22 +699,51 @@ func setupHealthzHandlers(ks *kube2sky) {
 	})
 }
 
+func validateArgs() error {
+	if *argEtcdServer == "" {
+		return fmt.Errorf("etcd server required")
+	}
+	if *argDomain == "" {
+		return fmt.Errorf("cluster domain required")
+	}
+	parsed, err := url.Parse(*argEtcdServer)
+	if err != nil {
+		return err
+	}
+	if parsed.Scheme == "https" && (*argEtcdCertFile == "" || *argEtcdKeyFile == "") {
+		return fmt.Errorf("both cert and key files must be provided for HTTPS")
+	}
+	// TODO: Validate other input flags.
+	return nil
+}
+
+func newContext() context.Context {
+	return kapi.WithNamespace(kapi.NewContext(), kube2skyContext)
```

Comment:
the context the etcd KeysAPI is looking for doesn't know anything about kubernetes namespaces